### DISCUSSION TITLE
Callable is not colored correctly

### DISCUSSION THREAD

**Discussion by pwall2222** [microsoft/pylance-release#3884] [Enhancement]
Callable gets highlighted as any other variable when in fact it is a callable so it would be great if it was highlighted like a function or method, or even just having the option to edit the color of it would be great. (This can't be impossible to do, maybe very hard, because typescript extension does work for this purpose)

<h3> Environment data </h3>

-   Language Server version: 2023.1.10
-   OS and version: win32 x64
-   Python version: 3.9.4 
-   python.analysis.indexing: null
-   python.analysis.typeCheckingMode: basic

<h3> Code Snippet </h3>

```python
from typing import Callable


def fn(sub_fn: Callable):
    return sub_fn()
```

<h3> Expected behavior </h3>

![image](https://user-images.githubusercontent.com/34860495/215364924-4f13d5b2-a0e1-49ff-8df5-8fd67e0b5437.png)

<h3> Actual behavior </h3>

![image](https://user-images.githubusercontent.com/34860495/215364807-6e3aec3b-f6d1-46e6-99eb-c2cff97166bb.png)

**Comment by erictraut** [comment]
`Callable` is a special form introduced in PEP 484. It's similar to a protocol class that defines a `__call__` method. I think it would be odd to color it the same as functions or methods declared with a `def` statement.

Incidentally, `Callable` is intended to be used with two type arguments. The first represents the parameter list and is either `...` (ellipsis) or a list of parameter types. The second represents the return type. In your screen shot, you have not provided either of these type arguments. A correct usage would be something like `Callable[..., int]` or `Callable[[int, str], str]`. You can see how it acts more like a class than an identifier that refers to a specific function or method.

**Comment by pwall2222** [comment]
Yea but even if you type it correctly, you still get the same color as if you had any other argument so if you pass it along to another method or you call it directly it doesn't tell you much by looking at it, it solely depends on the name you give it. I mean making it default to use the same color as any other function well maybe it's not the most friendly thing, but having the option to change it would at least be beneficial and up to the user to decide if they want to have the same color or not.

**Comment by heejaechang** [comment]
`sub_fn` is a parameter, so we color it as a parameter, same goes for variable `fn: Callable ...`, `fn` will be colored as a variable.

but that being said, we could attach token modifier for parameters, variables that are `callable` so user could customize them if they want to through [editor.semanticTokenColorCustomizations](https://code.visualstudio.com/docs/getstarted/themes%5C#_customizing-a-color-theme)

let's put it in the discussion to see user feedbacks.

**Reply by Chrysaloid** [reply]
Any updates on this?

If this is already working it would be nice if you provided a snippet for `settings.json` that would enable coloring of `Callable` parameters and variables.

So far I tried this:
```json
"editor.semanticTokenColorCustomizations": {
	"enabled": true,
	"rules": {
		"parameter.callable:python": { "foreground": "#dcdcaa" },
		"variable.callable:python": { "foreground": "#dcdcaa" },
		"parameter.Callable:python": { "foreground": "#dcdcaa" },
		"variable.Callable:python": { "foreground": "#dcdcaa" },
	}
},
```
...but it is not working :(

**Reply by rchiodo** [reply]
This isn't something that we've added support for. There's no extra data for callables yet. 

<img width="703" height="559" alt="image" src="https://github.com/user-attachments/assets/3dd80e7b-82c6-4339-b07f-4e12eae98b3f" />

You can see the data we return using `Developer : Inspect Editor Tokens and Scopes`

**Reply by Chrysaloid** [reply]
Thanks for the information. Let's hope someone will implement this feature someday...
